cluster-autoscaler/core/scaledown/unneeded/nodes.go

```diff
@@ -140,10 +131,12 @@ func (n *Nodes) updateInternalState(nodes []simulator.NodeToBeRemoved, ts time.T
 				ntbr:  nn,
 				since: val.since,
 			}
-		} else if updatedNode := updatedNodeBuilder(nn, ts); updatedNode != nil {
-			updated[name] = updatedNode
+		} else if ts := timestampGetter(nn, ts); ts != nil {
```

Comment:
nit: Since you're checking the return value for `nil`, you could skip passing `ts` in the argument to `timestampGetter` (just return `nil` instead of `&ts`).